File: VECTOAux/VectoAuxiliaries/Hvac/IBusDatabase.vb
Namespace Hvac

    Public Interface IBusDatabase

        Function AddBus(bus As IBus) As Boolean

        Function GetBuses(busModel As String, Optional AsSelectList As Boolean = False) As List(Of IBus)

        Function Initialise(busFileCSV As String) As Boolean

        Function UpdateBus(id As Integer, bus As IBus) As Boolean

        Function Save(filepath As String) As Boolean

    End Interface

End Namespace

File: VECTOAux/BusAuxiliaries/DownstreamModules/Impl/HVAC/IBusDatabase.cs
using System.Collections.Generic;

namespace TUGraz.VectoCore.BusAuxiliaries.DownstreamModules.Impl.HVAC
{
	public interface IBusDatabase
	{
		bool AddBus(IBus bus);

		List<IBus> GetBuses(string busModel, bool AsSelectList = false);

		bool Initialise(string busFileCSV);

		bool UpdateBus(int id, IBus bus);

		bool Save(string filepath);
	}
}
